equinox/nn/_batch_norm.py

```diff
@@ -10,7 +11,29 @@
 from ._stateful import State, StateIndex
 
 
-class BatchNorm(Module):
+class _StatefulLayer(Module):
+    """An abstract base class representing a stateful layer within a neural network.
+
+    Stateful layers maintain internal state across calls, which can be useful for tasks
+    such as BatchNormalisation.
+
+    Subclasses must implement the `__call__` method that takes input data and the
+    current state as arguments and returns the output data and updated state.
+    """
+
+    @abc.abstractmethod
+    def __call__(
+        self,
+        x,
+        state: State,
+        *,
+        key: Optional["jax.random.PRNGKey"] = None,  # pyright: ignore
```

Comment:
This should now be `Optional[jaxtyping.PRNGKeyArray]`, see the other examples elsewhere in Equinox.